lib/utils/angular/scroll_host/angular_2.dart

```diff
@@ -0,0 +1,323 @@
+
+  @override
+  Rectangle calcViewportRect() => _scrollHost.calcViewportRect();
+
+  @override
+  void scrollToPosition(int position) =>
+      _scrollHost?.scrollToPosition(position);
+
+  @override
+  void scrollWithDelta(int delta) => _scrollHost?.scrollWithDelta(delta);
+
+  @override
+  void startNativeScrollListener() => _scrollHost.startNativeScrollListener();
+
+  @override
+  int get scrollHeight => _scrollHost.scrollHeight;
+
+  @override
+  int get clientHeight => _scrollHost.clientHeight;
+
+  @override
+  Stream<ScrollHostEvent> get nativeOnScroll => _scrollHost.nativeOnScroll;
+
+  @override
+  bool get usePositionSticky => _scrollHost.usePositionSticky;
+
+  @override
+  bool get throttleScrollEvents => _scrollHost.throttleScrollEvents;
+
+  @override
+  GlobalEventHandlers get scrollbarHost => _scrollHost.scrollbarHost;
+
+  @override
+  int get clientWidth => _scrollHost.clientWidth;
+
+  @override
+  num get offsetX => _scrollHost.offsetX;
+
+  @override
+  num get offsetY => _scrollHost.offsetY;
+
+  @override
+  Element get anchorElement => _scrollHost.anchorElement;
+
+  @override
+  Stream<ScrollHostEvent> get onScroll => _scrollHost.onScroll;
+
+  @override
+  PanController get panController => _scrollHost.panController;
+
+  @override
+  StickyController get stickyController => _scrollHost.stickyController;
+
+  @override
+  int get scrollLength => _scrollHost.scrollLength;
+
+  @override
+  int get scrollPosition => _scrollHost.scrollPosition;
+
+  @override
+  Stream<IntersectionObserverEntry> onIntersection(Element element) =>
+      _scrollHost.onIntersection(element);
+
+  @override
+  void ngOnDestroy() {
+    dispose();
+  }
+}
+
+/// Provides a scroll host that uses the browser window content area.
+@Injectable()
+class WindowScrollHost extends WindowScrollHostBase implements OnDestroy {
+  WindowScrollHost(DomService domService, NgZone ngZone,
+      GestureListenerFactory gestureListenerFactory, Window window)
+      : super(domService, ngZone, gestureListenerFactory, window);
+
+  @override
+  ngOnDestroy() {
+    dispose();
+  }
+}
+
+/// A directive to track already sticking Elements on the page (e.g. one that
+/// has position fixed).
+///
+/// The sticky controller will respect their position and
+/// will put the other sticky headers and footers below or above them.
+@Directive(
+  selector: '[acxStickyFloating]',
+)
+class StickyFloatingTracker implements OnInit, OnDestroy {
+  final ScrollHost _scrollHost;
+  final Element _element;
+
+  StickyFloatingTracker(this._scrollHost, ElementRef elementRef)
+      : _element = elementRef.nativeElement;
+
+  @override
+  void ngOnInit() {
+    _scrollHost.stickyController.trackFloating(_element);
+  }
+
+  @override
+  void ngOnDestroy() {
+    _scrollHost.stickyController.untrackFloating(_element);
+  }
+}
+
+/// A directive to dynamically set a class based on the pan state.
+///
+/// Example use:
+///
+///     <some-element acxPanClass="'pan-style'" ...
```

Comment:
[PREFER backtick fences for code blocks.](https://www.dartlang.org/guides/language/effective-dart/documentation#prefer-backtick-fences-for-code-blocks)